addons/mrp/wizard/change_production_qty.py

```diff
@@ -102,9 +109,9 @@ def change_prod_qty(self):
                     wo.finished_lot_id = False
                     wo._workorder_line_ids().unlink()
                 wo.qty_producing = quantity
-                if wo.qty_produced < wo.qty_production and wo.state == 'done':
+                if wo.qty_produced < production_qty and wo.state == 'done':
                     wo.state = 'progress'
-                if wo.qty_produced == wo.qty_production and wo.state == 'progress':
+                if wo.qty_produced == production_qty and wo.state == 'progress':
```

Comment:
It would be nice if you add a quick test :angel:

Reply:
I've added a test. :)